### ISSUE TITLE
SpeechHistory 2020.2 stuck on "enabled after restart" status in Manage add-ons dialog of NVDA 2019.3.1

### ISSUE THREAD

**Issue by ScottChesworth** [jscholes/nvda-speech-history#12]
I'm attempting to use SpeechHistory 2020.2 with NVDA 2019.3.1. Installation appears to go smoothly, but once installed, its status is stuck on "enabled after restart" in the Manage add-ons dialog. Tried restarting NVDA multiple times, no change. Dunno if this is significant, but since installing SpeechHistory 2020.2, I get prompted to restart NVDA every time I close the Manage add-ons dialog even if I haven't made any changes that should require a restart. I'm also not seeing any trace of the new settings panel mentioned in the commit yet, so I guess SpeechHistory isn't running.

If you have any ideas of what might be up, I'd appreciate the insight.

**Comment by jscholes** [comment]
No idea I'm afraid.  Have you removed the add-on completely, and then tried installing it again?  I can't reproduce anything like that here.

**Comment by Ignite25** [comment]
Hi! I am  having the same problems as Mr.     Chesworth except I am using NVDA version 2020.1 Yes, I have installed and removed the add on multiple times. I downloaded  the Speech History add on  from Your Github site and the  NVDA community add ons website and tried them both. I did have this add on with earlier versions of NVDA before 19.3 and I liked it very much.

**Comment by jscholes** [comment]
If you can reliably reproduce the issue, I will need a debug log which follows the installation procedure of the add-on.  Otherwise I can't determine what's going wrong.

1. Uninstall Speech History and restart NVDA.
2. Set the logging level to debug in NVDA's General settings panel.
3. Restart NVDA again.
4. Install the latest version of Speech History.
5. Restart NVDA when prompted.
6. Attach the log file to a comment on this issue.  You can find it by:
    1. Pressing Windows+R.
    2. Typing "%temp%" (without the quotes) and pressing Enter.
    3. Locating the file nvda.log.

When you're done, set logging level back to info.

**Comment by ScottChesworth** [comment]
I've popped a log in Dropbox because apparently I'm not smart enough to figure out how to attach it to a comment here. Hope it's useful.
https://www.dropbox.com/s/l8lqhl27cezz8zg/nvda.log?dl=1

**Comment by jscholes** [comment]
Thanks for that.  It looks like for some reason, NVDA is failing to remove the old version of the add-on while trying to install the new one; it gives up and then tries to load the old version, which of course doesn't work.

If you uninstall Speech History, open C:\Users\Scott\AppData\Roaming\nvda\addons, and then ensure that the following two items are not present:

1. speechHistory.pendingInstall
2. speechHistory

Does that fix anything when you try to reinstall the latest version?

**Comment by Ignite25** [comment]
Yay! it works! Thanks Scott for uploading the debug log. I could do the first 5 steps James typed out  and I was just about to try the last step in which I didn't know if I could do. Then I checked my email and noticed you already uploaded a debug log. Saves me from trying. Anyways, the path James typed out didn't quite work for me so I typed in appdata in the windows run box and took it from there. Thanks James and Scott.
